color/src/missing.rs

```diff
@@ -10,21 +10,26 @@ use crate::x11_colors;
 /// This tracks missing color components of a `DynamicColor` and details of how a `DynamicColor`
 /// was constructed.
 ///
-/// The "missing" flags indicate whether a specific color component is missing. The "named" flag
-/// represents whether the dynamic color was generated from one of the named colors in [CSS Color
-/// Module Level 4 § 6.1][css-named-colors] or named color space functions in [CSS Color Module
-/// Level 4 § 4.1][css-named-color-spaces].
+/// The "missing" flags indicate whether a specific color component is missing (either the three
+/// color channels or the alpha channel). The "named" flag represents whether the dynamic color was
+/// generated from one of the named colors in [CSS Color Module Level 4 § 6.1][css-named-colors] or
+/// named color space functions in [CSS Color Module Level 4 § 4.1][css-named-color-spaces].
 ///
 /// The latter is primarily useful for serializing.
 ///
 /// [css-named-colors]: https://www.w3.org/TR/css-color-4/#named-colors
 /// [css-named-color-spaces]: https://www.w3.org/TR/css-color-4/#color-syntax
-//
-// The flags are tracked with 16 bits. The first three are for missing components, the fourth
-// indicates whether the color was generated from a named color or named color space function. The
-// remaining bits indicate the named color.
 #[derive(Default, Clone, Copy, PartialEq, Eq, Debug)]
-pub struct Flags(u16);
+pub struct Flags {
+    /// A bitset of missing color components.
+    missing: u8,
```

Comment:
Would it make sense for this to be `Missing` directly?

Reply:
Yes, I think so.

To shrink the API surface, maybe also remove the `missing` methods from `Flags`? Except for

```rust
pub const fn from_missing(missing: Missing) -> Self
pub fn set_missing(&mut self, missing: Missing)
pub fn missing(self) -> Missing
```

Code would then call as e.g. `dynamic_image.flags.missing().contains(1)`, and using the `Missing` bitops would fit more naturally into that pattern.

What are your thoughts?

Reply:
Seems reasonable enough.